src/test/java/com/google/sps/MealServletTest.java

```
@@ -0,0 +1,233 @@
+        2L, "Chocolate cake", "Chocolate cake with butter cream and strawberry.",
+            new ArrayList<>(Arrays.asList("flour", "water", "butter", "strawberry")), "Dessert");
+    
+
+
+    @Before
+    public void setUp() {
+        helper.setUp();
+    }
+
+    @After
+    public void tearDown() {
+        helper.tearDown();
+    }
+
+    // TODO(sandatsian): add more specific tests for search feature and get list feature.
+    // TODO(grenlayk): add test for get similar feature.
+    // Get one of the existing objects Meal by id.
+    @Test
+    public void getMealByIdTest() throws IOException, ServletException {
+        DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+        ds.put(createMealEntity(MEAL_1));
+        ds.put(createMealEntity(MEAL_2));
+        
+        MealServlet servlet = new MealServlet();
+        MockHttpServletRequest request = new MockHttpServletRequest();
+        MockHttpServletResponse response = new MockHttpServletResponse();
+        request.setPathInfo("/2");
+        
+        try {
+            servlet.doGet(request, response);
+        } catch(Exception e) {
+            System.out.println(e);
+        }
+        Gson gson = new Gson();
+        String expected = gson.toJson(MEAL_2);
+        String actual = response.getContentAsString().trim();
+
+        assertEquals(expected, actual);
+    }
+
+    // Get an object from empty datastore.
+    // Result: Response Status NOT FOUND.
+    @Test
+    public void getMealByIdFromEmptyDsTest() throws IOException, ServletException {
+        DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+        
+        MealServlet servlet = new MealServlet();
+        MockHttpServletRequest request = new MockHttpServletRequest();
+        MockHttpServletResponse response = new MockHttpServletResponse();
+        request.setPathInfo("/2");
+        
+        try {
+            servlet.doGet(request, response);
+        } catch(Exception e) {
```

In this scenario your `doGet()` isn't **expected** to throw exceptions. In such case, you have no need to `try {} catch {}` them. Therefore please remove these blocks and just call `servlet.doGet()` as usual.

If that throws an exception, it will be easier to analyse if you let it crash the test (instead of catching it).

Do this for every test.